src/lib/object.ts

```diff
@@ -0,0 +1,13 @@
+export const shake = <T>(obj: T): Shook<T> => {
+  if (obj == null) return obj
+  if (typeof obj !== 'object') return obj
+
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  const initialValue = {} as Shook<T>
```

Comment:
@phpnode I can't figure out how to do this without a type assertion. 🤷🏻

Reply:
you can't, TS isn't clever enough unfortunately.

Reply:
if we were being strict we'd say that this is a `Partial<T>` here, and it's only when we return that it's `as Shook<T>`, but I don't think it matters at all